src/commonMain/kotlin/io/ktor/io/Buffer.kt

```diff
@@ -265,6 +265,18 @@ public interface Buffer : Closeable {
         writeIndex += 8
     }
 
+    /**
+     * Split this [Buffer] into head and tail.
+     *
+     * The head buffer will have bytes between 0 and index inside. And the tail buffer will have bytes between [index]
+     * and [capacity]. The [readIndex] and [writeIndex] will be pointing to the same bytes in the result buffers.
+     *
+     * The head of split is returned and this buffer becomes the tail.
+     *
+     * @return head of split
+     */
+    public fun takeHead(index: Int = writeIndex): Buffer
```

Comment:
TBH, I'm not sure about the naming and overall it's contract.

Now we have:
1. absolute primitive operations (get/set)
2. relative primitive operations (read/write)
3. copy bulk(array/buffer) operations (both absolute and relative)

And this `takeHead`, which DETACHES HEAD part of current buffer from 0 up to some index looks not so good...
I see, that `netty5` have the same operation called `split`

Do you have some usecases for it's usage. I have some ideas, but it's hard for now to understand if it's will be useful.
BTW, why we (and netty) do return HEAD and not TAIL as result of this operation?

P.S. without experimenting with it it's hard for me to understand what is better